### PR TITLE
[SPARK-46317][PYTHON][CONNECT] Match minor behaviour matching in SparkSession with full test coverage

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR matches the corner case behaviours in `SparkSession` between Spark Connect and non-Spark Connect with adding unittests with the full test coverage within `pyspark.sql.session`.

### Why are the changes needed?

- For feature parity.
- To improve the test coverage.
    See https://app.codecov.io/gh/apache/spark/blob/master/python%2Fpyspark%2Fsql%2Fsession.py - this is not being tested.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Manually ran the new unittest.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: python/pyspark/errors/error_classes.py [hunk 1]
```diff
@@ -848,11 +848,6 @@
       "SparkContext or SparkSession should be created first.."
     ]
   },
-  "SHOULD_NOT_DATAFRAME": {
```
Comment (HyukjinKwon) [inline python/pyspark/errors/error_classes.py:851]: cc @zhengruifeng and @itholic

**Comment by HyukjinKwon** [comment]
Build https://github.com/HyukjinKwon/spark/actions/runs/7137173471

**Review by itholic — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

File: python/pyspark/sql/tests/test_session.py [hunk 2]
```diff
@@ -18,6 +18,8 @@
 import os
 import unittest
 import unittest.mock
+from io import StringIO
+from lxml import etree
```
Comment (dongjoon-hyun) [inline python/pyspark/sql/tests/test_session.py:22]: Oh, this seems to break the `PyPy` daily CI.
- https://github.com/apache/spark/actions/runs/7239278196/job/19721018105
```
Traceback (most recent call last):
  File "/usr/local/pypy/pypy3.8/lib/pypy3.8/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/pypy/pypy3.8/lib/pypy3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/__w/spark/spark/python/pyspark/sql/tests/test_session.py", line 22, in <module>
    from lxml import etree
ModuleNotFoundError: No module named 'lxml'
```

**Comment by dongjoon-hyun** [comment]
I made a follow-up for `lxml`.
- #44391

**Pull request closed without merging**